packages/client/test/end-to-end/erc1271-publish-subscribe.test.ts

```diff
@@ -0,0 +1,103 @@
+import { Wallet } from 'ethers'
+import { fastWallet, fetchPrivateKeyWithGas } from '@streamr/test-utils'
+import { waitForCondition, areEqualBinaries, toEthereumAddress, EthereumAddress } from '@streamr/utils'
+import { StreamrClient } from '../../src/StreamrClient'
+import { createTestStream, createTestClient } from '../test-utils/utils'
+import { StreamPermission } from '../../src/permission'
+import { deployMockERC1271Contract } from '../test-utils/deployMockERC1271Contract'
+import { StreamID } from '@streamr/protocol'
+import { MessageMetadata } from '../../src'
+
+const PAYLOAD = new Uint8Array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+const TIMEOUT = 30 * 1000
+
+describe('ERC-1271: publish and subscribe', () => {
```

Comment:
Reduce copy-paste by noticing that public stream / private stream test cases are essentially the same, only boolean `publicSubscribePermission` gets different value.

Reply:
done